### PR TITLE
Improve SignedInConfirmationDialog

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Improve `SignedInConfirmationDialog`.

#### HOW

- Align with Figma redlines;
- Display photo;
- Handle scenarios when any of the information is not available;
- Improve a11y for timeout;
- Add previews;
- Add snapshot tests;

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

File: auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt [hunk 1]
```diff
@@ -66,90 +86,93 @@ public fun SignedInConfirmationDialog(
...
-    displayName: String,
-    email: String,
-    modifier: Modifier = Modifier
+    modifier: Modifier = Modifier,
+    displayName: String? = null,
+    email: String? = null,
+    avatarUri: Any? = null
 ) {
+    val configuration = LocalConfiguration.current
+    val horizontalPadding = (configuration.screenWidthDp * 0.094).dp
```
Comment (yschimke) [inline auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt:106]: Where are the magic values from?

File: auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt [hunk 2]
```diff
@@ -66,90 +86,93 @@ public fun SignedInConfirmationDialog(
...
-    ) {
-        Column(
-            modifier = modifier.fillMaxSize(),
-            verticalArrangement = Arrangement.Center
-
-        ) {
+        if (hasEmail) {
             Text(
-                text = "Success!",
+                text = email!!,
```
Comment (yschimke) [inline auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt:167]: Do you avoid the need for !! if you inline hasEmail?

**Review by yschimke — APPROVED** [review]

File: auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt [hunk 3]
```diff
@@ -31,30 +32,49 @@ import androidx.compose.runtime.getValue
...
+    duration: Duration = Duration.ofMillis(DialogDefaults.ShortDurationMillis)
 ) {
     val currentOnDismissOrTimeout by rememberUpdatedState(onDismissOrTimeout)
 
+    val durationMillis = LocalAccessibilityManager.current?.calculateRecommendedTimeoutMillis(
+        originalTimeoutMillis = duration.toMillis(),
+        containsIcons = false,
+        containsText = true,
+        containsControls = false
+    ) ?: run {
```
[the remark this answers is not in the document]
Comment (luizgrp) [reply auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt:74]: removed, thanks 👍 left behind after changing code from `?.let`

File: auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt (hunk 1 above)
Comment (luizgrp) [reply auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt:106]: Figma redlines, added to consts

File: auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt (hunk 2 above)
Comment (luizgrp) [reply auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt:167]: not sure I follow, but removed `hasEmail` and used `email?.let`

Comment (luizgrp) [reply auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt]: I think they improve readability, even with the annoying double-bang, but happy to hear other perspectives. 

Compare:

```kotlin
avatarUri?.let {
    // content 1
} ?: run {
    displayName?.let {
        // content 2
    }
}
```

With:
```kotlin
if (hasAvatar) {
    // content 1
} else if (hasName) {
    // content 2
}
```

Comment (luizgrp) [reply auth-composables/src/main/java/com/google/android/horologist/auth/composables/dialogs/SignedInConfirmationDialog.kt]: also happy to hear any other alternative suggestion that you might have, merging the PR for now and will address it in a follow-up

**Pull request merged**